src/network/networkd-link.c

```diff
@@ -721,10 +721,15 @@ void link_check_ready(Link *link) {
                     !link->ipv4ll_route)
                         return;
 
-        if (link_ipv6ll_enabled(link))
+        if (link_ipv6_enabled(link)) {
+
                 if (in_addr_is_null(AF_INET6, (const union in_addr_union*) &link->ipv6ll_address) > 0)
                         return;
 
+                if (link_ipv6_accept_ra_enabled(link) && !link->ndisc_configured)
```

Comment:
This is a bit confusing IMHO, and the commit message doesn't quite match the patch: This change is not bridge specific at all. So `link_ipv6_accept_ra_enabled(link)` would be true even if `link_ipv6_enabled(link)` is false? In that case the patch  makes perfect sense and just the commit message should be clarified.

The commit message "drop the check for IPv6 settings" doesn't quite match the actual code here, and completely dropping it would also be wrong -- if you have a bridge where only the components get addresses but not the bridge interface itself, you actually *do* want to have `s-n-wait-online` wait for the bridge components. And indeed that mostly seems to be what your patch is doing, except that it replaces the check for "is link-layer address enabled" with "is RA enabled". Will the latter also be true for DHCPv6?